### PR TITLE
Add support for packed structs.

### PR DESCRIPTION
**Pull request by alvardes**
Fixes: #1747

### CODE REVIEW

**Comment by alvardes** [comment]
@dtolnay I force-pushed to trigger the new checks. Seems to pass.

File: serde_derive/Cargo.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ proc-macro = true
 [dependencies]
 proc-macro2 = "1.0"
 quote = "1.0"
-syn = { version = "1.0", features = ["visit"] }
+syn = { version = "1.0", features = ["visit", "extra-traits"] }
```
Comment (dtolnay) [inline serde_derive/Cargo.toml:29]: "extra-traits" is pretty expensive in compile time. Please avoid introducing a dependency on it.

Comment (alvardes) [reply serde_derive/Cargo.toml]: I replaced the use of `==` on `syn::AttrStyle` with a structural match.

**Comment by dtolnay** [comment]
Published in 1.0.110.

**Pull request merged**